setup.cfg

```diff
@@ -81,6 +81,51 @@ setup_requires =
     setuptools >= 38.4
     setuptools_scm
 
+[options.extras_require]
+io =
+    netCDF4
+    h5netcdf
+    scipy
+    pydap
+    zarr
+    fsspec
+    cftime
+    rasterio
+    cfgrib
+    ## Scitools packages & dependencies (for example: cartopy, cf-units) can be hard to install
+    # scitools-iris
+    ## Required by scitools-iris
+    # scitools-pyke
```

Comment:
```sh
~/src/sandbox/xarray enh/opt-extra*
venv ❯ pip install scitools-iris
...
~/src/sandbox/xarray enh/opt-extra* 33s
venv ❯ pip list                 
Package                Version  
---------------------- ---------
antlr4-python3-runtime 4.7.2    
Cartopy                0.18.0   
certifi                2020.6.20
cf-units               2.1.4    
cftime                 1.2.1    
cycler                 0.10.0   
dask                   2.29.0   
kiwisolver             1.2.0    
matplotlib             3.3.2    
netCDF4                1.5.4    
numpy                  1.19.2   
Pillow                 7.2.0    
pip                    19.2.3   
pyparsing              2.4.7    
pyshp                  2.1.2    
python-dateutil        2.8.1    
PyYAML                 5.3.1    
scipy                  1.5.2    
scitools-iris          2.4.0    
setuptools             41.2.0   
Shapely                1.7.1    
six                    1.15.0   
toolz                  0.11.1   
WARNING: You are using pip version 19.2.3, however version 20.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.


~/src/sandbox/xarray enh/opt-extra*
venv ❯ python -c 'import iris'                                                                                                                                                              
/home/avmo/src/sandbox/xarray/venv/lib/python3.8/site-packages/iris/config.py:84: UserWarning: Ignoring config item 'Resources':'test_data_dir' (section:option) as '/home/h05/itwl/projects/git/iris-test-data/test_data' is not a valid directory path.
  warnings.warn(msg.format(section, option, c_path))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/avmo/src/sandbox/xarray/venv/lib/python3.8/site-packages/iris/__init__.py", line 113, in <module>
    import iris.fileformats
  File "/home/avmo/src/sandbox/xarray/venv/lib/python3.8/site-packages/iris/fileformats/__init__.py", line 31, in <module>
    from . import netcdf
  File "/home/avmo/src/sandbox/xarray/venv/lib/python3.8/site-packages/iris/fileformats/netcdf.py", line 44, in <module>
    from pyke import knowledge_engine
ModuleNotFoundError: No module named 'pyke'

```

Reply:
It seems to be a recent addition (28 days ago) https://github.com/SciTools/iris/commit/4fab30a6a7373cf6a3c46007bcf6cbb4fac1bf15. I will remove it since they are working on a new release.